aspnetcore/signalr/streaming/sample/Hubs/StreamHub.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reactive.Linq;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR;
+
+namespace SignalRChat.Hubs
+{
+    public class StreamHub : Hub
+    {
+        public ChannelReader<int> Counter(int count, int delay)
+        {
+            var channel = Channel.CreateUnbounded<int>();
```

Comment:
I think I prefer this pattern (comment included), even though I know that's not what our current sample does. It makes it clearer that you're expected to give the Writer away to some other process to do the writing.

```csharp
public ChannelReader<int> Counter(int count, int delay)
{
    var channel = Channel.CreateUnbounded<int>();

    // We don't want to await WriteItems, otherwise we'd end up waiting 
    // for all the items to be written before returning the channel back to
    // the client.
    _ = WriteItems(channel.Writer, count, delay);

    return channel.Reader;
}

private async Task WriteItems(ChannelWriter<int> writer, int count, int delay)
{
    for (var i = 0; i < count; i++)
    {
        await channel.Writer.WriteAsync(i);
        await Task.Delay(delay);
    }

    channel.Writer.TryComplete();
}
```